### ISSUE TITLE
Better (corrected?) docs for affine parameters

### ISSUE THREAD

**Issue by matham** [Project-MONAI/MONAI#8726]
**Is your feature request related to a problem? Please describe.**

I've been using monai for 3d data augmentation and it's working great. However, the rotate/translate/scale parameters to e.g. `RandAffine` seem to expect a very specific input format, which is a bit counter-intuitive, which I think can benefit from better docs. From my observations,

Translation
-----

Translation translates in the negative direction. E.g. the following code passes:

```py
@pytest.mark.parametrize("offset", [-1, 1])
def test_monai_translate_input(offset):
    """
    This tests that the translate values expected by monai are negative of the
    real values.
    """
    data = torch.zeros((1, 5, 5, 5))
    data[0, 2, 2, 2] = 1

    affine = RandAffine(
        prob=1,
        translate_range=((-offset, -offset), 0, 0),
        cache_grid=True,
        spatial_size=(5, 5, 5),
        lazy=False,
    )
    transformed = affine(data, padding_mode="border")
    assert math.isclose(transformed[0, 2, 2, 2], 0)
    assert transformed[0, 2 + offset, 2, 2] > 0
```

Scaling
----

Scaling happens so that `(0, inf)` makes it smaller, while `[-1, 0)` makes it bigger:

```py
@pytest.mark.parametrize(
    "interval,growth",
    [
        ((100, 100), "smaller"),
        ((-1, -1), "bigger"),
        ((100, 0), "not bigger"),
        ((0, -1), "not smaller"),
    ]
)
def test_monai_scale_input(interval, growth):
    """
    This tests that the unexpected scale interval is indeed expected by monai.
    """
    data = torch.zeros((1, 5, 5, 5))
    data[0, 1:4, 1:4, 1:4] = 1

    affine = RandAffine(
        prob=1,
        scale_range=(interval,) * 3,
        cache_grid=True,
        spatial_size=(5, 5, 5),
        lazy=False,
    )
    transformed = affine(data, padding_mode="border")

    match growth:
        case "smaller":
            # it got scaled down, check edge became zero
            assert math.isclose(transformed[0, 3, 3, 3], 0)
        case "bigger":
            # it got scaled up, check outside became non-zero
            assert transformed[0, 4, 4, 4] > 0
        case "not bigger":
            # it got either smaller or stayed the same, check outside stayed zero
            assert math.isclose(transformed[0, 4, 4, 4], 0)
        case "not smaller":
            # it got either bigger or stayed the same, check edge stayed non-zero
            assert transformed[0, 3, 3, 3] > 0
```

Rotation
-----

Rotation seems to happen clockwise around the given axis (actually only around x and y. Around 1 it's counter-clockwise):

```py
@pytest.mark.parametrize("angle", [math.pi / 2, 3 * math.pi / 2])
def test_monai_rotate_input(angle):
    """
    This tests the angle values expected by monai is normal, but clockwise.
    """
    data = torch.zeros((1, 5, 5, 5))
    data[0, 2, 2, 1] = 1

    # rotate around the first axis
    affine = RandAffine(
        prob=1,
        rotate_range=((angle, angle), 0, 0),
        cache_grid=True,
        spatial_size=(5, 5, 5),
        lazy=False,
    )
    transformed = affine(data, padding_mode="border")

    assert math.isclose(transformed[0, 2, 2, 1], 0)
    if angle < math.pi:
        assert transformed[0, 2, 1, 2] > 0
    else:
        assert transformed[0, 2, 3, 2] > 0
```

**Describe the solution you'd like**

Better documentation that clearly specifies the expected input and its meaning. The current input seems counterintuitive so I'm not sure if I'm doing something work or if this is just how it is expected. But if the latter, then docs saying so would be appreciated.
